### PR TITLE
Ensure we use the correct tag while composing a milestone Publish Name

### PR DESCRIPTION
**Pull request by adamfarley**
Documentation tweak. See modified doc for details.

Signed-off-by: Adam Farley <adam.farley@uk.ibm.com>

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -154,6 +154,7 @@ JDK_FIX_VERSION=0
    * `targetConfigurations`: remove all "hotspot" entries
    * `releaseType`: `Nightly`
    * `overridePublishName`: github binaries publish name, e.g. `jdk8u232-b09_openj9-0.17.0-m1` or `jdk-11.0.5+10_openj9-0.17.0-m1`
+   (Note: Everything before the underscore should be copied from the OPENJDK\_TAG value inside <extensions_repo_url>/closed/openjdk-tag.gmk)
```
Comment (keithc-ca) [inline RELEASING.md:157]: Why the backslash in `OPENJDK\_TAG` (I don't see other underscores escaped)?

Comment (adamfarley) [reply RELEASING.md:157]: In markdown documents, anything after an underscore is displayed in _italics_. I didn't want italics, but I did want an underscore.

Comment (adamfarley) [reply RELEASING.md]: Backslash removed.

**Review by sxa — APPROVED** [review]
Seems ok for me - and nice catch on the 11 vs 8 error in there - nice to know someone was checking it ;-)

**Review by karianna — APPROVED** [review]
LGTM

**Pull request merged**